src/kakarot/evm.cairo

```diff
@@ -144,32 +145,52 @@ namespace EVM {
     // @param amount The amount of gas the current operation requires.
     // @return EVM The pointer to the updated execution context.
     func charge_gas{range_check_ptr}(self: model.EVM*, amount: felt) -> model.EVM* {
-        let out_of_gas = is_le_felt(self.gas_left + 1, amount);
+        tempvar a = self.gas_left - amount;
+        %{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
```

Comment:
add code comment explaining what's done here and referring to is_le_felt as a reference

Reply:
Will do 
Also note that the inlining, is the equivalent of doing `is_nn(amount - self.gas_left)` in the reference implementation in the issue [here](https://github.com/kkrt-labs/kakarot/issues/1261).

Reply:
done [b3a38a5](https://github.com/kkrt-labs/kakarot/pull/1323/commits/b3a38a5018083a098259327658632f2191c0d007)